File: frontend2/Rugby/src/app/classes/game.ts
import {Opponent} from './opponent';
import {AthleteGameStats} from './associations/AthleteGameStats';
import {Athlete} from './athlete';
import { ActiveRoster } from './associations/ActiveRoster';

export class Game {
  constructor(
    public id?: number,
    public date?: Date,
    public local?: string,
    public comment?: string,
    public opponent?: Opponent,
    public athletes?: Athlete[],
    public activeRoster?: ActiveRoster[]
  ) {
    this.id = id ? id : 0;
    this.comment = comment ? comment : '';
    this.date = date ? date : new Date(0);
    this.local = local ? local : '';
    this.opponent = opponent ? opponent : new Opponent();
    this.athletes = athletes ? athletes : [];
    this.activeRoster = activeRoster ? activeRoster : [];
  }
}

File: backend/src/main/java/pt/isel/rugby/model/Game.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "game")
@Data
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private Date date;

    @Column
    private String local;

    @Column
    private String comment;

    @OneToOne
    @JoinColumn(name = "opponentId", referencedColumnName = "id")
    private Opponent opponent;

    @ManyToMany(mappedBy = "games")
    private List<Athlete> athletes;

    @OneToMany(mappedBy = "game" , cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ActiveRoster> activeRoster = new ArrayList<>();

    /*
    public void addActiveRoster(ActiveRoster ar){
        activeRoster.add(ar);
        ar.setGame(this);
    }

    public void removeActiveRoster(ActiveRoster ar){
        activeRoster.remove(ar);
        ar.setGame(null);
    }*/

    @OneToMany(mappedBy = "game")
    @JsonIgnore
    private List<AthleteGameStats> athleteGameStats;

    @ManyToOne
    @JsonIgnore
    private Tournament tournament;
}
